back-end/db/player.ts

```diff
@@ -53,9 +64,9 @@ export default {
     }
   },
 
-  getPlayerByName: async (playerName: string): Promise<any> => {
+  getPlayerByName: async (playerName: string, gameId: number): Promise<any> => {
     try {
-      const player = await Player.findOne({name: playerName});
+      const player = await Player.findOne({name: playerName, gameId: gameId});
```

Comment:
Need to pass game ID here - otherwise, it could grab a player with the same name from a different game